### PR TITLE
Add table.project evaluator

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
This PR
- adds evaluator for table.project operator
- updates the parser to interpret "Table" as a table with zero columns

### CODE REVIEW

File: src/theory/datatypes/tuple_utils.h [hunk 1]
```diff
@@ -33,6 +33,22 @@ class TupleUtils
...
+  /**
+   * @param indices a list of indices for projected elements
+   * @param tuple a node of tuple type
+   * @return the projection of the tuple with the specified indices
+   */
+  static Node getTupleProjection(const std::vector<uint32_t>& indices,
+                                 Node tuple);
+
+  /**
+   * @param indices  a list of indices for projected elements
```
Comment (ajreynol) [inline src/theory/datatypes/tuple_utils.h:45]:
```suggestion
   * @param indices a list of indices for projected elements
```

**Review by ajreynol — CHANGES_REQUESTED** [review]
This looks nice.

I think we need to throw a LogicException if we see TABLE_PROJECT here: https://github.com/cvc5/cvc5/blob/main/src/theory/bags/theory_bags.cpp#L455

File: src/theory/bags/table_project_op.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+#include "theory/datatypes/tuple_project_op.h"
+
+namespace cvc5::internal {
+
+/**
+ * The class is an operator for kind project used to project elements in a
+ * table. It stores the indices of projected elements
+ */
+class TableProjectOp : public TupleProjectOp
```
[the remark this answers is not in the document]
Comment (mudathirmahgoub) [reply src/theory/bags/table_project_op.h:29]: These classes are identical. I tried to reuse TupleProjectOp in bags, but it conflicts with auto-generated code. The best approach is to allow such feature and rename TupleProjectOp to ProjectOp as you suggested. Later, RelationProjectOp would be similar.

[the remark this answers is not in the document]
Comment (mudathirmahgoub) [reply src/theory/bags/table_project_op.h]: Added ProjectOp class to tuple_project_op.h

**Comment by mudathirmahgoub** [comment]
> This looks nice.
> 
> I think we need to throw a LogicException if we see TABLE_PROJECT here: https://github.com/cvc5/cvc5/blob/main/src/theory/bags/theory_bags.cpp#L455

Done

**Review by ajreynol — APPROVED** [review]

**Pull request merged**